### PR TITLE
OpenCVE

### PR DESCRIPTION
**Pull request by rfortress-pan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
This integration pulls CVEs from an instance of OpenCVE for enhanced enrichments.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py [hunk 1]
```diff
@@ -0,0 +1,1013 @@
...
+        raise Exception('Please provide a valid value for the Source Reliability parameter.')
+
+    # Instantiate the OpenCVE object
+    ocve = OpenCVE(url=url, username=username, password=password,
+                   verify_ssl=verify_ssl, tlp=tlp, reliability=reliability)
+
+    command = demisto.command()
+    args = demisto.args()
+
+    LOG(f'Command being called is {command}')
```
Comment (ilaner) [inline Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:958]: please replace with demisto.info

File: Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py [hunk 2]
```diff
@@ -0,0 +1,1013 @@
...
+
+
+def debug(msg, url='https://demo.xsoar.engineer/nodered', endpoint='test'):
+    url = f'{url}/{endpoint}'
+    if type(msg) in [dict, list]:
+        return requests.post(url, json=msg)
+    return requests.post(url, data=f'{msg}')
+
+
+class OpenCVE():
```
Comment (ilaner) [inline Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:18]: Consider inheriting from `BaseClient` of `CommonServerPython`:
https://xsoar.pan.dev/docs/tutorials/tut-integration-ui

File: Packs/OpenCVE/.pack-ignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[file:DemistoClassApiModule.py]
```
Comment (ilaner) [inline Packs/OpenCVE/.pack-ignore:1]: No need to include it

**Review by ilaner — CHANGES_REQUESTED** [review]
Thank you for your contribution!
Great job, please see the comments and the build errors (lint and validate)

File: Packs/OpenCVE/pack_metadata.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "OpenCVE",
+    "description": "Ingests CVEs from OpenCVE",
+    "support": "xsoar",
```
Comment (ilaner) [inline Packs/OpenCVE/pack_metadata.json:4]: please change to community

**Review by ilaner — CHANGES_REQUESTED** [review]
Thank you for your contribution.
Please see the comments.
In addition, please resize the integration image size to 120x50.

File: Packs/OpenCVE/pack_metadata.json (hunk 4 above)
Comment (rfortress-pan) [reply Packs/OpenCVE/pack_metadata.json]: updated

File: Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py (hunk 1 above)
Comment (rfortress-pan) [reply Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py]: replaced

File: Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py (hunk 2 above)
Comment (rfortress-pan) [reply Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:11]: considered

**Comment by ilaner** [comment]
Please add descriptions to the commands and fix the errors seen here
[Details](https://circleci.com/gh/demisto/content/663887)

**Comment by ilaner** [comment]
@rfortress-pan 
There are still erros in validations:
```
[2023-07-12 09:08:14,127] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-vendor-cves'-> 'outputs'-> 'vendor_cves'
[2023-07-12 09:08:14,128] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-vendor-cves'-> 'outputs'-> 'vendor_cves' in the yml.

[2023-07-12 09:08:14,128] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-reports'-> 'outputs'-> 'reports'
[2023-07-12 09:08:14,128] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-reports'-> 'outputs'-> 'reports' in the yml.

[2023-07-12 09:08:14,129] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-report'-> 'arguments'-> 'report_id'
[2023-07-12 09:08:14,129] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-report'-> 'arguments'-> 'report_id' in the yml.

[2023-07-12 09:08:14,129] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-report'-> 'outputs'-> 'report'
[2023-07-12 09:08:14,129] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-report'-> 'outputs'-> 'report' in the yml.

[2023-07-12 09:08:14,130] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-alerts'-> 'arguments'-> 'report_id'
[2023-07-12 09:08:14,130] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-alerts'-> 'arguments'-> 'report_id' in the yml.

[2023-07-12 09:08:14,130] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-alerts'-> 'outputs'-> 'alerts'
[2023-07-12 09:08:14,130] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-alerts'-> 'outputs'-> 'alerts' in the yml.

[2023-07-12 09:08:14,130] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-alert'-> 'arguments'-> 'report_id'
[2023-07-12 09:08:14,131] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-alert'-> 'arguments'-> 'report_id' in the yml.

[2023-07-12 09:08:14,131] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-alert'-> 'arguments'-> 'alert_id'
[2023-07-12 09:08:14,131] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-alert'-> 'arguments'-> 'alert_id' in the yml.

[2023-07-12 09:08:14,131] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-alert'-> 'outputs'-> 'alert'
[2023-07-12 09:08:14,132] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-alert'-> 'outputs'-> 'alert' in the yml.

[2023-07-12 09:08:14,132] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'arguments'-> 'cve_id'
[2023-07-12 09:08:14,132] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'arguments'-> 'cve_id' in the yml.

[2023-07-12 09:08:14,132] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.ID'
[2023-07-12 09:08:14,133] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.ID' in the yml.

[2023-07-12 09:08:14,133] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.CVSS'
[2023-07-12 09:08:14,133] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.CVSS' in the yml.

[2023-07-12 09:08:14,133] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.Published'
[2023-07-12 09:08:14,133] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.Published' in the yml.

[2023-07-12 09:08:14,134] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.Modified'
[2023-07-12 09:08:14,134] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.Modified' in the yml.

[2023-07-12 09:08:14,134] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.Description'
[2023-07-12 09:08:14,134] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'CVE.Description' in the yml.

[2023-07-12 09:08:14,135] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Indicator'
[2023-07-12 09:08:14,135] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Indicator' in the yml.

[2023-07-12 09:08:14,135] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Score'
[2023-07-12 09:08:14,135] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Score' in the yml.

[2023-07-12 09:08:14,135] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Type'
[2023-07-12 09:08:14,136] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Type' in the yml.

[2023-07-12 09:08:14,136] [ERROR] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.yml: [ST107] - Missing the field "description" in Path: 'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Vendor'
[2023-07-12 09:08:14,136] [INFO] Please add the field "description" to the path:  'script'-> 'commands'-> 'ocve-get-cve'-> 'outputs'-> 'DBotScore.Vendor' in the yml.


Validating Packs/OpenCVE/Integrations/OpenCVE/OpenCVE_image.png as image
[2023-07-12 09:08:14,158] [INFO] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE_image.png: [IM101] - Too large logo, please update the logo to be under 10kB

[2023-07-12 09:08:14,158] [INFO] Packs/OpenCVE/Integrations/OpenCVE/OpenCVE_image.png: [IM111] - The image dimensions are 500x150. The requirements are 120x50.
```

```
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:369:19: error:
Item "None" of "Optional[Any]" has no attribute "replace"  [union-attr]
            published=cve.get('timestamp').replace('Z', ''),
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:370:18: error:
Item "None" of "Optional[Any]" has no attribute "replace"  [union-attr]
            modified=cve.get('modified').replace('Z', '')
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:389:64: error:
Invalid index type "str" for "str"; expected type "Union[SupportsIndex, slice]" 
[index]
    ...            cve_context[field_item] = parsed_cve['fields'][field_item]
                                                                  ^~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:430:9: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
            parsed_cve['fields']['tags'] = tags
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:447:21: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
                        parsed_cve['fields']['publications'] = references
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:454:17: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
                    parsed_cve['fields']['cvssscore'] = cvssV2.get('baseSc...
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:455:17: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
                    parsed_cve['fields']['cvssvector'] = ocve._map(cvssV2....
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:456:17: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
                    parsed_cve['fields']['cvsstable'] = [
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:471:17: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
                    parsed_cve['fields']['cvssscore'] = cvssV3.get('baseSc...
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:472:17: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
                    parsed_cve['fields']['cvssvector'] = ocve._map(cvssV3....
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:473:17: error:
Unsupported target for indexed assignment
("Union[Dict[str, Any], str, Any, None]")  [index]
                    parsed_cve['fields']['cvsstable'] = [
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:483:12: error:
Incompatible return value type (got
"Dict[str, Union[Dict[str, Any], str, Any, None]]", expected "Dict[str, str]") 
[return-value]
        return parsed_cve
               ^~~~~~~~~~
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:531:1: error:
Missing return statement  [return]
    def test_module_command(client: Client) -> str:
    ^
/home/circleci/project/Packs/OpenCVE/Integrations/OpenCVE/OpenCVE.py:541:9: error:
"Client" has no attribute "get_my_vendors"; maybe "get_my_vendors_request"? 
[attr-defined]
            client.get_my_vendors()
            ^~~~~~~~~~~~~~~~~~~~~
Found 14 errors in 1 file (checked 1 source file)
```
https://app.circleci.com/pipelines/github/demisto/content/281063/workflows/13cf3f85-9893-4350-9832-449cf0b1936c

I fixed some of the errors, but take a look on the remaining errors. We can schdule a meeting to help you sort them out.

**Review by ilaner — APPROVED** [review]
Good job!

**Pull request merged**